src/vm/database/marf.rs

```diff
@@ -14,56 +18,67 @@ use chainstate::burn::BlockHeaderHash;
 ///   loop will need to invoke these two methods (begin + commit) outside of the context of the VM.
 ///   NOTE: Clarity will panic if you try to execute it from a non-initialized MarfedKV context.
 ///   (See: vm::tests::with_marfed_environment()) 
-pub struct MarfedKV <S: KeyValueStorage> {
+pub struct MarfedKV {
     chain_tip: BlockHeaderHash,
-    marf: MARF,
+    // the MARF is option'ed, if None, then this KV is "marfless".
+    //    this is used for raw evals, testing --- use cases formerly that used a generic datastore.
+    // functions which _assume_ a MARF present will panic if the MARF is None.
+    marf: Option<MARF>,
     // Since the MARF only stores 32 bytes of value,
     //   we need another storage
-    side_store: S
+    side_store: SqliteConnection
 }
 
-#[cfg(test)]
-pub fn temporary_marf() -> MarfedKV<std::collections::HashMap<String, String>> {
+struct ContractCommitment {
+    pub hash: [u8; 32],
+    pub block_height: u32
+}
+
+impl ContractCommitment {
+    pub fn serialize(&self) -> String {
+        format!("{}{}", to_hex(&self.hash), to_hex(&self.block_height.to_be_bytes()))
+    }
+    pub fn deserialize(input: &str) -> ContractCommitment {
```

Comment:
Not a bug, but would there be an advantage to having this method return a `Result` and having the higher-level database trying to load it call `.expect` so it can at least provide more error context?

Reply:
The other database serialization and deserialization functions used for the side storage all panic, so this was implemented to match that. These serialization functions aren't used otherwise at the moment (the ones that are, like Value deserialization, do have Result returning functions), so I don't think there's much value in adding a Result type here, since the only caller will panic on it anyways.